x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -705,6 +701,14 @@ public List<RoleDescriptor> parseRoleDescriptorsBytes(
         final String apiKeyId,
         BytesReference bytesReference,
         RoleReference.ApiKeyRoleType roleType
+    ) {
+        return parseRoleDescriptorsBytes(apiKeyId, bytesReference, roleType == RoleReference.ApiKeyRoleType.LIMITED_BY);
+    }
+
+    private List<RoleDescriptor> parseRoleDescriptorsBytes(
+        final String apiKeyId,
+        BytesReference bytesReference,
+        final boolean replaceLegacySuperuserRoleDescriptor
```

Comment:
This was actually the initial version that I had for this method. It got changed to using an Enum for better semantics. Since this is now a private method, I think it is OK.

Reply:
Agreed that the flag is suitable for a private method whereas enum is better for public API 👍